File: JavaScript/Remove Duplicates from Sorted List.js
// https://leetcode.com/problems/remove-duplicates-from-sorted-list/

/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 *
 * Given 1->1->1->2->3->3, return 1->2->3.
 */
var deleteDuplicates = function(head) {
  if (!head) return []
  var node = head
  while (node && node.next) {
    while(node.next && node.next.val == node.val) {
      node.next = node.next.next
    }
    node = node.next
  }
  return head
};

File: Go/083-remove-duplicates-from-sorted-list/remove_duplicates_from_sorted_list.go
package deleteDuplicates

type ListNode struct {
	Val  int
	Next *ListNode
}

func deleteDuplicates(head *ListNode) *ListNode {
	for curr := head; curr != nil; curr = curr.Next {
		for curr.Next != nil && curr.Val == curr.Next.Val {
			curr.Next = curr.Next.Next
		}
	}
	return head
}
